ecleankernel/file.py

```diff
@@ -97,8 +102,10 @@ def __init__(self,
         super().__init__(path, KernelFileType.KERNEL)
         self.internal_version = self.read_internal_version()
 
-    def decompress_raw(self) -> bytes:
-        f = open(self.path, 'rb')
+    def ver_from_raw(self, f: BinaryIO) -> Union[str, Any]:
```

Comment:
I think `IO[bytes]` is more general. Also, what's the point of union with `Any`?

Reply:
I am not really sure how this type ended up here, but I see that if it's only `str` then we get typing failures for some of the returns of decompressed data, e.g. the returns of this code:

```
                if comp == 'zstandard':
                    # Technically a redundant import, this is just
                    # to make your IDE happy :)
                    import zstandard
                    reader = zstandard.ZstdDecompressor().stream_reader(f)
                    decomp = b''
                    while True:
                        chunk = reader.read(1024 * 1024)
                        if not chunk:
                            break
                        decomp += chunk
                    return decomp
                elif comp == 'lzma':
                    # Using .decompress() causes an error because of
                    # no end-of-stream marker
                    return LZMADecompressor().decompress(f.read())
```

I suppose we actually only want strings returned from this function, so shall I just `str()` them?